deploy/runtools/utils.py

```diff
@@ -360,10 +362,11 @@ def get_local_shared_libraries(elf):
     ]
 
     libs = list()
-    rootLogger.debug(f"Identifying ldd dependencies for:{elf}")
+    rootLogger.debug(f"Identifying ldd dependencies for: {elf}")
     for dso in lddwrap.list_dependencies(Path(elf)):
         if dso.soname is None:
-            assert '/ld-linux' in fspath(dso.path), f"dynamic linker is only allowed no soname, not: {dso}"
+            if dso.path is not None:
```

Comment:
@timsnyder-siv the type checker errors here since dso.path could potentially be `None`. What should this be?

Reply:
When I checkout your `runtools-mypy` branch and `./scripts/run-py-typecheck.sh`, I don't see an error.  What am I missing?

Reply:
> IDK why I can't comment on this: https://github.com/firesim/firesim/pull/1042#discussion_r859030911. But I added the extra if dso.path is not None: check. Is that correct?

I'm only able to comment in the thread if I go to the "Files Changed" tab, not in Conversations for some reason.

I looked back though lddwrap and I think I really meant 
```
       assert dso.path is not None and '/ld-linux' in fspath(dso.path), ...
```

I don't want it to silently continue past anything other than the dynamic linker in that case.